packages/ember-metal/lib/properties.ts

```diff
@@ -188,16 +217,18 @@ export function defineProperty(
   if (desc instanceof Descriptor) {
     value = desc;
 
-    Object.defineProperty(obj, keyName, {
-      configurable: true,
-      enumerable,
-      get: DESCRIPTOR_GETTER_FUNCTION(keyName, value),
-    });
-
-    meta.writeDescriptors(keyName, value);
-
-    if (typeof desc.setup === 'function') {
-      desc.setup(obj, keyName);
+    if (desc instanceof NativeDescriptor) {
+      Object.defineProperty(obj, keyName, desc.desc);
+    } else {
+      Object.defineProperty(obj, keyName, {
+        configurable: true,
+        enumerable,
+        get: DESCRIPTOR_GETTER_FUNCTION(keyName, value),
+      });
+      meta.writeDescriptors(keyName, value);
```

Comment:
The `meta.writeDescriptors` here was previously done in both cases and now its only done for non-native.

Reply:
Yes sorry, was going to ask if that is bad idea, else will investigate, tests don’t fail though

Reply:
To be honest, I am unsure if it is important, but I would assume that we don't want to change the results of things like `descriptorFor` and whatnot.